subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -25,10 +25,27 @@ static struct download_client   dlc;
 static struct k_delayed_work    dlc_with_offset_work;
 static int socket_retries_left;
 
+static void send_evt(enum fota_download_evt_id id)
+{
+	const struct fota_download_evt evt = {
```

Comment:
`assert(id != FOTA_DOWNLOAD_EVT_PROGRESS);`?

Reply:
Someone told me that we shouldn't use asserts in a library only build asserts but that could be wrong.

Reply:
Added assert